### PR TITLE
fix(ses): Error cause permits

### PR DESCRIPTION
**Pull request by kriskowal**
The cause property on errors is now a stage 4 proposal and in evidence in the wild.

### CODE REVIEW

File: packages/ses/NEWS.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 User-visible changes in SES:
 
+# Next release
+
+- Updates the permits to include the `cause` accessor on all prototypes.
```
Comment (mhofman) [inline packages/ses/NEWS.md:5]:
```suggestion
- Updates the permits to include `cause` placeholders on Error prototypes.
```

**Review by mhofman — CHANGES_REQUESTED** [review]
AFAIK, there are no implementations where `cause` is an accessor on the Error prototype, just a dummy `undefined` value in v8, probably to prevent the error object instances from shifting shape.

File: packages/ses/src/whitelist.js [hunk 2]
```diff
@@ -546,6 +547,7 @@ export const whitelist = {
     constructor: '%SharedError%',
     message: 'string',
     name: 'string',
+    cause: 'undefined',
```
Comment (mhofman) [inline packages/ses/src/whitelist.js:550]:
```suggestion
    cause: false,
```

Apparently it was a [bug](https://github.com/tc39/notes/blob/master/meetings/2021-10/oct-26.md#:~:text=However%2C%20Chrome%2093,and%20node%2016.11.) in v8's implementation to have included it in the first place:
> However, Chrome 93 and node 16.9 and 16.10 shipped with `cause` property on `Error.prototype` as well, which is not supposed to be there. It's technically allowed by the spec, but that was not the design of the feature or the intention and it's not what the other engines have chosen to do. And so, of course, that obvious bug was fixed in Chrome 94 and node 16.11.

**Comment by kriskowal** [comment]
Notes: when I get back to this issue, the plan of record is to delete these properties in the repair phase instead of permitting them.

**Comment by kriskowal** [comment]
I’ve posted a fresh attempt.

**Review by mhofman — APPROVED** [review]
I don't claim to understand the whitelist mechanism, but why isn't setting to `false` there not sufficient?

**Comment by kriskowal** [comment]
@mhofman That is in fact the better solution. Posted a different commit to address the same issue, this time using a permit.

**Review by erights — APPROVED** [review]
LGTM

**Pull request merged**